src/main/java/it/gov/pagopa/pu/send/service/SendFacadeServiceImpl.java

```diff
@@ -135,8 +138,29 @@ public SendNotificationDTO notificationStatus(String sendNotificationId) {
     return sendNotificationDTOMapper.apply(notification);
   }
 
+  @Override
+  public NotificationPriceResponseV23DTO retrieveNotificationPrice(Long organizationId, String iuv) {
+    SendNotificationNoPII notification = findSendNotificationByOrgIdAndIUV(organizationId, iuv);
+
+    // Validate status
+    NotificationUtils.validateStatus(NotificationStatus.ACCEPTED, notification.getStatus());
+    Payment payment = notification.getPayments().stream()
+      .map(PuPayment::getPayment)
+      .filter(pagoPa -> iuv.equals(pagoPa.getPagoPa().getNoticeCode().substring(1)))
```

Comment:
move into a separate common method iuv2nav
```suggestion
      .filter(pagoPa -> iuv.equals(iuv2nav(pagoPa.getPagoPa().getNoticeCode())))
```